### DISCUSSION TITLE
File Listing

### DISCUSSION THREAD

**Discussion by Godlike-Productions** [RomanKubiak/ctrlr#464] [Q&A] [answered]
Does anyone know how to get this code working?

https://ctrlr.org/forums/topic/advice-before-starting-the-project/page/3/#post-3664

or another way to get a list of files in a directory into a table?

The juce file operations don't seem to have bindings to lua

**Comment by dnaldoog** [comment, marked as answer]
This slight adaption of [that ](https://ctrlr.org/forums/topic/advice-before-starting-the-project/page/3/#post-3664) code works!

<pre>
myMethod = function(mod, value, source)
    local t = {}
    local f = File("c:\\") -- create a file that points to the directory we want to list
    f:findChildFiles(t, File.findDirectories, false, "*")
    -- search the file for directories only, don’t recurse
    for i, v in ipairs(t) do
        console(string.format("[%u] %s", i, v:getFullPathName()))
    end -- print each found element in the table
end

</pre>

**Reply by Godlike-Productions** [reply]
Thankyou so much

**Reply by dobo365** [reply]
John is always fast to answer ;-)
Was going to answer that one as well as I'm using that for my coming Super 6 editor and librarian.
Greetings, guys!

**Reply by dnaldoog** [reply]
Time zone advantage! Lol
